docs/current_docs/tensors.md

```diff
@@ -142,6 +142,131 @@ GaussianVertex vertex = new GaussianVertex(shape, mu, 0);
  */
 ```  
 
+## Tensor Broadcasting
+
+Broadcasting enables you to perform operations across tensors of different shape, rank and length.
+
+* length (the total number of elements)
+* rank (the number of dimensions)
+* shape (the length in each dimension). 
+
+Tensors can be added, subtracted, multiplied and more with each other. 
+It wouldn’t be very useful if you could only operate on tensors of the same shape, fortunately you can perform broadcasting.  
+
+### Tensor and Scalar
+
+Let’s start with a simple example, multiplying each element inside a matrix by a constant.
+
+Let’s define A as a tensor of shape [2, 2]. It’s therefore a 2x2 matrix of rank 2. 
+Let’s define B as a tensor of shape []. It’s therefore a rank 0 constant (a scalar).
+
+```java
+DoubleTensor A = DoubleTensor.create(new double[]{
+    10., 20.,
+    30., 40.
+}, 2, 2); //shape [2, 2]
+
+DoubleTensor B = DoubleTensor.scalar(5.); //shape []
+
+DoubleTensor C = A.times(B); //shape [2, 2]
+```
+
+We can do operations as normal between the two, even though they have different shapes. 
+In the case of a tensor and a scalar, the resulting tensor will always have the shape of the initial tensor.
+
+
+### Tensor and Tensor
+
+You will want to operate on tensors of different sizes. 
+Fortunately this is also supported given that you abide by certain broadcasting rules.
+
+Numpy have a very clear and thorough explanation of when broadcasting between tensors is valid and explanations of 
+how we determine which dimension to broadcast along that can be found [here](https://docs.scipy.org/doc/numpy-1.15.0/user/basics.broadcasting.html).
+
+In summary: When analysing two tensor shapes to determine if they are broadcastable, always read from the right. 
+If they are different rank, just ignore the remaining values on the left. 
+
+There is one simple rule that must be abided for broadcasting to be successful. 
+
+* When reading the shapes from the right, do they match or is one of the values a 1?
+
+Here are some examples of both cases.
+
+#### Subset of shape
+
+*Valid Example*
+
+Operating on a `[2, 2, 2]` and a `[2, 2]`
+
+```java
+DoubleTensor A = DoubleTensor.create(new double[]{
+    10., 20., 30., 40.,
+    50., 60., 70., 80.
+}, 2, 2, 2);  //shape [2, 2, 2]
+
+DoubleTensor B = DoubleTensor.arange(1, 8).reshape(2, 2);  //shape [2, 2]
+
+DoubleTensor C = A.times(B); //shape [2, 2, 2]
+DoubleTensor D = A.plus(B); //shape[2, 2, 2]
```

Comment:
For clarity of what the operation is doing, you should print out the results

Reply:
This section is really just focusing on which shapes are broadcastable. It's a good point though - I've added the results of the operation to the complete example just below this one.